src/conn.c

```diff
@@ -3333,6 +3346,18 @@ natsConnection_Connect(natsConnection **newConn, natsOptions *options)
     return NATS_UPDATE_ERR_STACK(s);
 }
 
+natsStatus
+natsConnection_Reconnect(natsConnection *nc)
+{
+    natsStatus s = NATS_OK;
+
+    if (natsConnection_IsClosed(nc))
+        return nats_setDefaultError(NATS_INVALID_ARG);
+
+    IFOK(s, _forceReconnect(nc, NATS_OK, NULL));
```

Comment:
There is no need for the `IFOK` here since there is no change of `s` in lines above.

But overall, I think these changes are way more complicated than it needs to be (or I am missing something). If we want to cause a reconnect, locking the connection and and if the socket is valid, closing it should be enough, no?

Of course, verification that the connection is set in such a way that it allows for reconnection (I don't recall if there are options that would totally prevent a reconnect) before doing so, and if not, returning an error indication that reconnect() failed because the connection would not reconnect because of options setting, would be best.

Reply:
> way more complicated than it needs to be (or I am missing something). If we want to cause a reconnect, locking the connection and and if the socket is valid, closing it should be enough, no?

@kozlovic you're likely right, I was modeling after the `go` changes that call an existing re-connect method, and I figured this to be the closest. I'll try what you suggested, in a separate branch.

Reply:
See https://github.com/nats-io/nats.c/pull/757